src/DynamoCoreWpf/ViewModels/Search/NodeSearchElementViewModel.cs

```diff
@@ -233,14 +241,85 @@ public ImageSource LargeIcon
         /// <summary>
         /// Create the search element as node and connect to target port
         /// </summary>
-        public ICommand CreateAndConnectCommand { get; private set; }
+        public ICommand CreateAndConnectCommand { get; }
 
-        public event Action<string, PortModel> CreateAndConnectToPort;
-        protected virtual void OnRequestCreateAndConnectToPort(PortModel portModel)
+        /// <summary>
+        /// Create new node for search element, connect to port and place using graph auto layout.
+        /// </summary>
+        /// <param name="parameter">Port model to connect to</param>
+        protected virtual void CreateAndConnectToPort(object parameter)
         {
-            if (CreateAndConnectToPort != null)
+            var portModel = (PortModel) parameter;
+            var dynamoViewModel = searchViewModel.dynamoViewModel;
+
+            // Initialize a new undo action group before calling 
+            // node CreateAndConnect and AutoLayout commands.
+            if (undoRecorderGroup == null)
             {
-                CreateAndConnectToPort(Model.CreationName, portModel);
+                undoRecorderGroup = dynamoViewModel.CurrentSpace.UndoRecorder.BeginActionGroup();
+
+                // Node auto layout can be performed correctly only when the positions and sizes
+                // of nodes are known, which is possible only after the node views are ready.
+                dynamoViewModel.NodeViewReady += AutoLayoutNodes;
+            }
+
+            var initialNode = portModel.Owner;
+            var initialNodeVm = dynamoViewModel.CurrentSpaceViewModel.Nodes.FirstOrDefault(x => x.Id == initialNode.GUID);
+            var id = Guid.NewGuid();
+
+            var adjustedX = initialNodeVm.X;
+
+            var createAsDownStreamNode = portModel.PortType == PortType.Output;
+            // Placing the new node based on which input port it is connecting to.
+            if (createAsDownStreamNode)
+            {
+                // Placing the new node to the right of initial node
+                adjustedX += initialNode.Width + 50;
+            }
+            else
+            {
+                // Placing the new node to the left of initial node
+                adjustedX -= initialNode.Width + 50;
+            }
+
+            // Create a new node based on node creation name and connection ports
+            dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(id, initialNode.GUID,
+                Model.CreationName, 0, portModel.Index, adjustedX, 0, createAsDownStreamNode, false, true));
+
+            // Clear current selections and select all input nodes as we need to perform Auto layout on only the input nodes.
+            DynamoSelection.Instance.ClearSelection();
```

Comment:
@aparajit-pratap Do you know if selection is part of undo/redo sequence? I have been thinking about if there could be any side effects to this but in general feels OK

Reply:
This is not part of the undo/redo sequence. I think you'll need to use one of the selection related commands to add it to undo/redo queue.